_posts/blog/2017-08-21-google-assistant-development-for-android-devs-part-2.md

```diff
@@ -0,0 +1,112 @@
+
+    fun suggestions(app: ApiAiApp) {
+        app.ask(app
+            .buildRichResponse()
+            .addSimpleResponse("This is a simple response for suggestions")
+            .addSuggestions("Suggestion Chips")
+            .addSuggestions("Basic Card", "List", "Carousel")
+            .addSuggestionLink("Suggestion Link", "https://assistant.google.com/"))
+    }
+    
+    val actionMap = mapOf(
+        WELCOME to ::welcome,
+        NORMAL_ASK to ::normalAsk,
+        SUGGESTIONS to ::suggestions)
+      
+    
+    @WebServlet("/conversation")
+    class WebHook : HttpServlet() {
+
+    	override fun doPost(req: HttpServletRequest, resp: HttpServletResponse) {
+        	ApiAiAction(req, resp).handleRequest(actionMap)
+       }
+    }
+```
+{% endraw %}
+
+### Actions in Java
+
+{% raw %}
+```java
+	@WebServlet("/conversation/java")
+	public class ConversationComponentsSampleJava extends HttpServlet {
+    	private static final Logger logger = Logger.getAnonymousLogger();
+
+		Function1<ApiAiApp, Object> welcome = app -> {
+        	app.ask(app.buildRichResponse()
+                .addSimpleResponse("Hi there from Java!", "Hello there from Java!")
+                .addSimpleResponse(
+                        "I can show you basic cards, lists and carousels as well as suggestions on your phone",
+                        "I can show you basic cards, lists and carousels as well as suggestions")
+                .addSuggestions("Basic Card", "List", "Carousel", "Suggestions"), null);
+        	return Unit.INSTANCE;
+    	};
+
+    	Function1<ApiAiApp, Object> normalAsk = app ->
+       	     app.ask("Ask me to show you a list, carousel, or basic card");
+
+    	Function1<ApiAiApp, Object> suggestions = app ->
+       	     app.ask(app.buildRichResponse(null)
+                    .addSimpleResponse("This is a simple response for suggestions", null)
+                    .addSuggestions("Suggestion Chips")
+                    .addSuggestions("Basic Card", "List", "Carousel")
+                    .addSuggestionLink("Suggestion Link", "https://assistant.google.com/"));
+
+		private Map<String, Function1<String, Object>> intentMap = new HashMap() {{
+        	put(ConversationComponentsSampleKt.WELCOME, welcome);
+        	put(ConversationComponentsSampleKt.NORMAL_ASK, normalAsk);
+        	put(ConversationComponentsSampleKt.SUGGESTIONS, suggestions);
+    	}};
+
+    	@Override
+    	protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+         	 ApiAiAction action = new ApiAiAction(req, resp);
+       	 	 action.handleRequest(intentMap);
+    	}
+    }
+```
+{% endraw %}
+
+ 
+Many Android and Java developers should be able to get started quickly, even if your're a newbie to back-end development.  Check out [our GitHub repo]((https://github.com/TicketmasterMobileStudio/actions-on-google-kotlin) for examples and more information on getting started.
```

Comment:
- your're -> you're (the typo that wouldn't die! haha)
- You have an extra open paren in the URL, which is making the link break
- Maybe also add a `{:target='_github'}` to this URL and the one earlier up?